src/usecases/validateCreateCompany.js

```diff
@@ -0,0 +1,28 @@
+import ValidationError from "../errors/validationError.js";
+
+function validateCreateCompany(company) {
+  const cpfCnpjRegex =
+    /(^\d{3}\.\d{3}\.\d{3}\-\d{2}$)|(^\d{2}\.\d{3}\.\d{3}\/\d{4}\-\d{2}$)/;
+
+  const listErrrors = [];
+
+  for (const key in company) {
+    if (!company[key]) {
+      listErrrors.push(`O campo ${key} é obrigatório`);
+    }
+  }
```

Comment:
Achei uma solução bastante perspicaz hehe. Mas, na prática, ela não assegura que os campos obrigatórios serão validados, ela só se baseia nas chaves que são enviadas no request body. Então se eu não enviar o `name`, por exemplo, ele nunca vai ser validado, pois como ele não existe no objeto, consequentemente ele não vai existir nas chaves do objeto. 

Para seguir com essa solução, aconselho você criar uma lista com os parâmetros que são obrigatórios, e com base na lista de parâmetros obrigatórios você valida o objeto company. Exemplo:
```suggestion
  const requiredKeys = ['name', 'document', 'city', 'image'];
  for (const key in requiredKeys) {
    if (!company[key]) {
      listErrrors.push(`O campo ${key} é obrigatório`);
    }
  }
```

Dessa forma, você consegue validar se o objeto `company` possui os campos obrigatórios.

Reply:
Sua sugestão foi mais perspicaz ainda querido! hehe

Tá alterado, não me atentei a esse detalhe.